docs/plugins/integrating-search-into-plugins.md

```diff
@@ -333,3 +333,85 @@ reading the search context.
 benefit. Issues and pull requests welcome.
+
+#### Custom search results
+
+Search results throughout Backstage are rendered as lists so that list items can easily be customized; although a default result list item is available, plugins are in the best position to provide custom result list items that surface relevant information only known to the plugin.
+
+The example below imagines `YourCustomSearchResult` as a type of search result that contains associated `tags` which could be rendered as chips below the title/text.
+
+```tsx
+import { Link } from '@backstage/core-components';
+import { useAnalytics } from '@backstage/core-plugin-api';
+import { ResultHighlight } from '@backstage/plugin-search-common';
+import { HighlightedSearchResultText } from '@backstage/plugin-search-react';
+
+type CustomSearchResultListItemProps = {
+  result: YourCustomSearchResult;
+  rank?: number;
+  highlight?: ResultHighlight;
+};
+
+export const CustomSearchResultListItem = (
+  props: CustomSearchResultListItemProps,
+) => {
+  const { title, text, location, tags } = props.result;
+
+  const analytics = useAnalytics();
+  const handleClick = () => {
+    analytics.captureEvent('discover', title, {
+      attributes: { to: location },
+      value: props.rank,
+    });
+  };
+
+  return (
+    <Link noTrack to={location} onClick={handleClick}>
+      <ListItem alignItems="center">
+        <Box flexWrap="wrap">
+          <ListItemText
+            primaryTypographyProps={{ variant: 'h6' }}
+            primary={
+              highlight?.fields?.title ? (
+                <HighlightedSearchResultText
+                  text={highlight.fields.title}
+                  preTag={highlight.preTag}
+                  postTag={highlight.postTag}
+                />
+              ) : (
+                title
+              )
+            }
+            secondary={
+              highlight?.fields?.text ? (
+                <HighlightedSearchResultText
+                  text={highlight.fields.text}
+                  preTag={highlight.preTag}
+                  postTag={highlight.postTag}
+                />
+              ) : (
+                text
+              )
+            }
+          />
+          {tags &&
+            tags.map((tag: string) => (
+              <Chip key={tag} label={`Tag: ${tag}`} size="small" />
+            ))}
+        </Box>
+      </ListItem>
+      <Divider />
+    </Link>
+  );
+};
+```
+
+The optional use of the `<HighlightedSearchResultText>` component makes it possible to highlight relevant parts of the result based on the user's search query.
+
+**Note on Analytics**: In order to track and improve search experiences across Backstage, it's important to understand when and what users search for, as well as what they click on after searching. When implementing a custom result component, it's your responsibility to instrument it according to search analytics conventions. In particular:
```

Comment:
do we need to mention having analytics set up in the app?

Reply:
Probably not necessary (since app-level setup is optional), but that reminds me to think about audiences reading through this documentation (plugin devs) vs. who this benefits (app integrators).  ...Will tweak the language a bit to better reflect that, and I'll link off to the plugin analytics docs for good measure.

Reply:
https://github.com/backstage/backstage/pull/15361/commits/9da74dcee627826c310c63e28065b1d1099fb035